src/components/cards/collaborations/AddCollaboratorButton.jsx

```diff
@@ -0,0 +1,232 @@
+  const [selectedUserGuid, setSelectedUserGuid] = useState(null);
+
+  const {
+    data: users,
+    isLoading: isUsersLoading,
+    error: getUsersError,
+  } = useGetUsers();
+
+  const {
+    mutate: addCollaboration,
+    isLoading: isAddCollaborationLoading,
+    error: addCollaborationError,
+    reset: resetAddCollaborationMutation,
+  } = mutation;
+
+  const error = addCollaborationError || getUsersError;
+
+  function handleCloseDialog() {
+    setIsDialogOpen(false);
+    setSearchTerm('');
+    setSelectedUserGuid(null);
+    resetAddCollaborationMutation();
+  }
+
+  function handleSubmitSearchTerm(e) {
+    e.preventDefault();
+
+    const form = e.target;
+    const formData = new FormData(form);
+    const searchValue = formData.get('search');
+
+    setSearchTerm(searchValue);
+    setSelectedUserGuid(null);
+  }
+
+  function handleSubmitCollaborator() {
+    addCollaboration(
+      { userGuid: selectedUserGuid },
+      { onSuccess: handleCloseDialog },
+    );
+  }
+
+  const currentCollaboratorGuids = (
+    userData?.collaborations || []
+  ).map(collaboration => {
+    const memberGuids = Object.keys(collaboration?.members || {});
+    return memberGuids.find(guid => guid !== userGuid);
+  });
+
+  const columns = [
+    { name: 'name', labelId: 'NAME', align: 'left' },
+    { name: 'email', labelId: 'EMAIL', align: 'left' },
+    { name: 'role', labelId: 'ROLE', align: 'left' },
+    {
+      name: 'created',
+      labelId: 'CREATED',
+      align: 'left',
+      options: {
+        cellRenderer: cellRendererTypes.date,
+        cellRendererProps: { fancy: true },
+      },
+    },
+  ];
+
+  const tableData = (users || [])
+    .filter(user =>
+      filterCollaborator(
+        user,
+        searchTerm,
+        userGuid,
+        currentCollaboratorGuids,
+      ),
+    )
+    .map(user => ({
+      guid: user?.guid,
+      email: user?.email,
+      name: user?.full_name,
+      created: user?.created,
+      role: intl.formatMessage({ id: getHighestRoleLabelId(user) }),
+    }));
+
+  return (
+    <>
+      <StandardDialog
+        titleId="ADD_COLLABORATION"
+        open={isDialogOpen}
+        onClose={handleCloseDialog}
+        fullWidth
+        maxWidth="md"
+      >
+        <DialogContent
+          style={{
+            display: 'flex',
+            flexDirection: 'column',
+            gap: 12,
+          }}
+        >
+          {users && (
+            <form
+              style={{
+                display: 'flex',
+                alignItems: 'center',
+                gap: 12,
+              }}
+              onSubmit={handleSubmitSearchTerm}
+            >
+              <TextField
+                name="search"
+                variant="outlined"
+                fullWidth
+                autoFocus
+                label={
+                  <FormattedMessage id="SEARCH_USER_INSTRUCTION" />
+                }
+                InputProps={{ startAdornment: <SearchIcon /> }}
```

Comment:
should InputProps be inputProps? Does case matter?

Reply:
According to the [docs](https://v4.mui.com/api/text-field/#props), case does matter:
- **inputProps** - Attributes applied to the input element.
- **InputProps** - Props applied to the Input element. It will be a FilledInput, OutlinedInput or Input component depending on the variant prop value.

`startAdornment` is a prop for a React component, not an attribute for a DOM element.

Reply:
... this is so weird haha. Good to "know".